example/e2e/helpers.ts

```diff
@@ -0,0 +1,218 @@
+    await element(by.id(Selectors.EXPIRY_DATE_INPUT)).typeText(props.expiry);
+    await element(by.id(Selectors.SECURITY_CODE_INPUT)).typeText(props.code);
+    await element(
+      by.id(Selectors.PAY_NOW_BUTTON).and(by.traits(['button']))
+    ).tap();
+  } else if (device.getPlatform() === 'android') {
+    await element(by.id(Selectors.ANDROID_CARD)).replaceText(props.number);
+    await element(by.id(Selectors.ANDROID_NAME)).replaceText(props.name);
+    await element(by.id(Selectors.ANDROID_EXPIRY)).replaceText(props.expiry);
+    await element(by.id(Selectors.ANDROID_CODE)).typeText(props.code);
+    await element(by.id(Selectors.ANDROID_PAY_NOW)).tap();
+    await device.enableSynchronization();
+  }
+}
+
+export async function assertResultsScreen(props: Props) {
+  await waitFor(element(by.text('Result')))
+    .toExist()
+    .withTimeout(15000);
+  await expect(element(by.id('receiptId-value'))).not.toHaveText('');
+  await expect(element(by.id('type-value'))).toHaveText(props.type);
+  await expect(element(by.id('result-value'))).toHaveText(props.result);
+}
+
+export async function complete3DS2() {
+  if (device.getPlatform() === 'android') {
+    await waitFor(element(by.text(Selectors.THREEDS2_TITLE_ANDROID)))
+      .toBeVisible()
+      .withTimeout(15000);
+    await new Promise((resolve) => setTimeout(resolve, 1500));
+    try {
+      for (let i = 0; i < 3; i++) {
+        await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress();
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      }
+    } catch (exception) {}
+  } else {
+    await waitFor(element(by.text(Selectors.THREEDS2_SCREEN_HEADER)))
+      .toBeVisible()
+      .withTimeout(15000);
+    await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress();
+    await new Promise((resolve) => setTimeout(resolve, 5000));
+    await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress();
+  }
+}
+
+export async function isCardAddedToPaymentMethods(): Promise<boolean> {
+  try {
+    await expect(element(by.label(Selectors.EXISTING_CARD))).toBeVisible();
+    return true;
+  } catch (exception) {
+    return false;
+  }
+}
+
+export async function addCardPaymentMethodAndPay() {
+  if (await isCardAddedToPaymentMethods()) {
+    await element(by.label(Selectors.EXISTING_CARD)).swipe('left');
+    await element(by.text('Delete')).tap();
+    await new Promise((resolve) => setTimeout(resolve, 3000));
+  }
+  if (device.getPlatform() === 'android') {
+    await element(by.text(Selectors.ADD_CARD_BUTTON)).tap();
+  } else {
+    await element(by.text(Selectors.ADD_CARD_BUTTON)).atIndex(1).tap();
+  }
+  await fillPaymentDetailsSheet({
+    number: TestData.CARD_NUMBER,
+    name: TestData.CARDHOLDER_NAME,
+    expiry: TestData.EXPIRY_DATE,
+    code: TestData.SECURITY_CODE,
+  });
+  if (device.getPlatform() === 'android') {
+    await new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
Since you use this in a few places to delay things, maybe it's worth `extracting` this to a separate function like so:

```javascript
function delay(milliseconds) {
    return new Promise(resolve => {
        setTimeout(() => {
            resolve();
        }, milliseconds);
    });
}
```

And then:

```javascript
await delay(2000)
```

Reply:
Thanks for this - refactored